### PR TITLE
🍒 Fix back to dashboard to select correct tab and preserve filters

### PR DESCRIPTION
**Pull request by uladzimirdev**
Backport #32192
and a fix from https://github.com/metabase/metabase/pull/32315

### CODE REVIEW

**Review by kamilmielnik — APPROVED** [review]

**Comment by kamilmielnik** [comment]
@uladzimirdev FYI I re-run 2 failed jobs here.

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js [hunk 1]
```diff
@@ -256,8 +284,42 @@ describe(
...
+        cy.wait("@dashcardQuery");
+        cy.get("@dashcardQuery.all").should("have.length", 1);
       });
+
+      filterWidget().findByPlaceholderText("sleep").type("5{enter}");
+
+      cy.wait("@dashcardQuery");
+      cy.get("@dashcardQuery.all").should("have.length", 2);
+
+      filterWidget().findByPlaceholderText("sleep").should("have.value", "5");
```
Comment (kamilmielnik) [inline e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:307]: This

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js [hunk 2]
```diff
@@ -256,8 +284,42 @@ describe(
...
+      queryBuilderHeader().findByLabelText("Back to Sleep dashboard").click();
+
+      filterWidget().findByPlaceholderText("sleep").should("have.value", "5");
+      getDashboardCard().within(() => {
+        cy.findByText("Sleep card").should("be.visible");
+        cy.get(".cellData").should("be.visible").and("contain", 5);
+      });
+
+      cy.log("Dashcard data should not be re-requested");
+      cy.get("@dashcardQuery.all").should("have.length", 2);
```
Comment (kamilmielnik) [inline e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:322]: The assertion is the opposite of what landed in `master`: https://github.com/metabase/metabase/pull/32192/files#diff-ab4bbcb5d3e1ba388de75ed1dcad298ad17ab278ac588837b56ee59ac51a313aR314-R315

Is this intentional?


Also, it does not match the test name.

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js [hunk 3]
```diff
@@ -256,8 +284,42 @@ describe(
...
-        cy.findByText("0").should("be.visible");
+        cy.findByText("Sleep card").should("be.visible");
+      });
+    });
+
+    it("should preserve filter value when navigating between the dashboard and the question", () => {
+      createDashboardWithSlowCard();
+      cy.get("@dashboardId").then(id => {
+        // Explicitly not using `visitDashboard` helper here because it also
+        // sets an intecept for the same route `POST /api/dashboard/*/dashcard/*/card/*/query`
```
Comment (kamilmielnik) [inline e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:295]:
```suggestion
        // sets an intercept for the same route `POST /api/dashboard/*/dashcard/*/card/*/query`
```

**Comment by uladzimirdev** [comment]
testing the fix here, if it works I want to merge it to master first and then cherry pick together with other changes in the 47 https://github.com/metabase/metabase/pull/32315

**Pull request merged**